src/filesystem/filesystem.ts

```diff
@@ -0,0 +1,7 @@
+export interface NotebookFilesystem {
+  get(absolutePath: string): Promise<string>;
```

Comment:
I would propose to make all of these take an options object instead (so that in the future we can add more opts, like the 'mode' when writing, and potentially also filesystem-specific information if we have to).

As for returning: I think it makes sense if it would return something like
```typescript
type AsyncResult<T,E> = Promise<{
  ok: true,
  data: T
} | {
 ok: false,
 status: number, // http-like status
 error: E,
 detail?: string
}>
```

I would also rename `absolutePath` to `path`. When we start making composable filesystems you could have a *prefix filesystem* which prefixes input paths with a prefix (so the user doesn't have to care about it).

Reply:
Huh, apparently one really cannot set the error return type of a promise using Typescript. As in, when you have something like `myPromise.catch((err) => {})`, the `err` type will always be a rather useless `any`

I did add a comment to the `NotebookFilesystem` about what error type it can return? Let me know if that's good enough

Reply:
That's why I'm proposing to use the AsyncResult type and aim for that it never throws. Instead one checks the `ok` field, and if it's not `ok` we will know the error type (and nature with 404). One can plaster/proxy over this with an API that does throw of course (which may be what people want as it's more succint to write)